src/CodeGen_Xtensa.cpp

```diff
@@ -1424,6 +1571,28 @@ HALIDE_ALWAYS_INLINE native_vector_u16_x3 halide_xtensa_interleave_u16(const nat
       58, 0,  22, 1,  23, 48, 59, 2,  24, 3,  25, 49, 60, 4,  26, 5,
       27, 50, 61, 6,  28, 7,  29, 51, 62, 8,  30, 9,  31, 52, 63, 10};
   unsigned long long int_16B_c3_step_1_msk = 0xffffffff55555555ULL;
+  #elif XCHAL_VISION_TYPE == 8
```

Comment:
Is this also from programming guide?

Reply:
Yes

Reply:
These aren't declared as 'static', so we are creating a new instance every time we call this function -- is the overhead significant?